ssl.go

```diff
@@ -61,39 +61,73 @@ func (fm *Frontman) runSSLCheck(addr *net.TCPAddr, hostname, service string) (m
 	defer connection.Close()
 
 	m[prefix+"expiryDaysRemaining"] = math.MaxFloat64
-	for _, cert := range connection.ConnectionState().PeerCertificates {
-		remainingValidity := time.Until(cert.NotAfter).Hours() / 24
+	certs := connection.ConnectionState().PeerCertificates
 
-		if remainingValidity < m[prefix+"expiryDaysRemaining"].(float64) {
-			m[prefix+"expiryDaysRemaining"] = remainingValidity
+	for i, cert := range certs {
+		opts := x509.VerifyOptions{
+			Intermediates: x509.NewCertPool(),
+		}
+		if !cert.IsCA && hostname != "" {
+			opts.DNSName = hostname
 		}
 
-		if remainingValidity <= 0 {
-			err = fmt.Errorf("certificate is expired: %s", certName(cert))
-			return
-		} else if remainingValidity <= float64(fm.Config.SSLCertExpiryThreshold) {
-			err = fmt.Errorf("certificate will expire soon: %s", certName(cert))
-			return
+		if i == 0 {
+			for j, cert := range certs {
+				if j != 0 {
+					opts.Intermediates.AddCert(cert)
+				}
+			}
 		}
 
-		if cert.NotBefore.After(time.Now()) {
-			err = fmt.Errorf("certificate is not valid yet: %s", certName(cert))
+		var chains [][]*x509.Certificate
+		chains, err = cert.Verify(opts)
```

Comment:
"cert" is used here and it is unclear if it is intended to be the "cert" set on line 66 or possibly line 74. Avoiding reuse of variable name would make the intention more obvious.